### ISSUE TITLE
Reporting

### ISSUE THREAD

**Issue by michmoor0725** [AdguardTeam/AdGuardHome#6071]
### Prerequisites

- [X] I have checked the [Wiki](https://github.com/AdguardTeam/AdGuardHome/wiki) and [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions) and found no answer

- [X] I have searched other issues and found no duplicates

- [X] I want to request a feature or enhancement and not ask a question

### The problem

Have adguard provide a reporting mechanism to see an individual or all clients and their DNS requets for a given time frame. Data is already in the database so now its about presenting data for it to be actionable in some way.

### Proposed solution

I imagine the report would contain a client and all the dns requests they made in a given time frame and how many times each dns request was requested.
The report will have simple selections for $host , $range , $time. Or maybe filtering based on threats/blocked/parent controls.
Basically an easier and cleaner way of viewing the Query log.

**Comment by michmoor0725** [comment]
Just want to add. Having a report and seeing youtube.com or roku.com showing up as top results for a bunch of clients makes it easier to write custom rules targeting clients.
